File: sample-github/src/main/java/com/omegar/mvp/sample/github/di/AppComponent.java
package com.omegar.mvp.sample.github.di;

import android.content.Context;

import com.omegar.mvp.sample.github.di.modules.BusModule;
import com.omegar.mvp.sample.github.di.modules.ContextModule;
import com.omegar.mvp.sample.github.di.modules.GithubModule;
import com.omegar.mvp.sample.github.mvp.GithubService;
import com.omegar.mvp.sample.github.mvp.presenters.RepositoriesPresenter;
import com.omegar.mvp.sample.github.mvp.presenters.RepositoryLikesPresenter;
import com.omegar.mvp.sample.github.mvp.presenters.SignInPresenter;
import com.squareup.otto.Bus;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Date: 8/18/2016
 * Time: 14:49
 *
 * @author Artur Artikov
 */
@Singleton
@Component(modules = {ContextModule.class, BusModule.class, GithubModule.class})
public interface AppComponent {
	Context getContext();
	GithubService getAuthService();
	Bus getBus();

	void inject(SignInPresenter presenter);
	void inject(RepositoriesPresenter repositoriesPresenter);
	void inject(RepositoryLikesPresenter repositoryLikesPresenter);
}

File: sample-app/src/main/kotlin/example/com/androidxsample/inject/AppComponent.kt
package example.com.androidxsample.inject

import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.DaggerApplication
import example.com.androidxsample.App
import javax.inject.Singleton

@Singleton
@Component(modules = [AppBindingModule::class])
interface AppComponent : AndroidInjector<DaggerApplication> {
	
	override fun inject(app: DaggerApplication)
	
	@Component.Builder
	interface Builder {
		
		@BindsInstance
		fun application(application: App): Builder
		
		fun build(): AppComponent
	}
	
}